qutebrowser/browser/qutescheme.py

```diff
@@ -234,21 +255,49 @@ def history_iter(start_time, reverse=False):
         except ValueError as e:
             raise QuteSchemeError("Query parameter start_time is invalid", e)
 
-        if sys.hexversion >= 0x03050000:
-            # On Python >= 3.5 we can reverse the ordereddict in-place and thus
-            # apply an additional performance improvement in history_iter.
-            # On my machine, this gets us down from 550ms to 72us with 500k old
-            # items.
-            history = history_iter(start_time, reverse=True)
-        else:
-            # On Python 3.4, we can't do that, so we'd need to copy the entire
-            # history to a list. There, filter first and then reverse it here.
-            history = reversed(list(history_iter(start_time, reverse=False)))
-
-        return 'text/html', json.dumps(list(history))
+        return 'text/html', json.dumps(history_data(start_time))
     else:
-        return 'text/html', jinja.render('history.html', title='History',
-                session_interval=config.get('ui', 'history-session-interval'))
+        if (
+            config.get('content', 'allow-javascript') and
+            objects.backend == usertypes.Backend.QtWebEngine
```

Comment:
You can also use the new page if `objects.backend == usertypes.Backend.QtWebKit and qtutils.is_qtwebkit_ng(qWebKitVersion())` (however you'll need to do a conditional import for `qWebKitVersion()`)